source/MaterialXCore/Types.h

```diff
@@ -292,6 +305,23 @@ template <class M, class S, size_t N> class MatrixN : public MatrixBase
     /// Return true if the given vector differs from this one.
     bool operator!=(const MatrixN& rhs) const { return _arr != rhs._arr; }
 
+    /// Return true if the given matrix is equivalent to another
+    /// matrix within a given floating point tolerance
+    bool equivalent(const MatrixN& rhs, float tolerance)
```

Comment:
I would recommend "isEquivalent" for the name of method, for consistency with the verb/noun form used in most of the MaterialX library.